src/init.cpp

```diff
@@ -1610,34 +1609,6 @@ bool AppInitMain(const CoreContext& context, NodeContext& node, interfaces::Bloc
         StartScriptCheckWorkerThreads(script_threads);
     }
 
-    {
-        LOCK(activeMasternodeInfoCs);
-        assert(activeMasternodeInfo.blsKeyOperator == nullptr);
-        assert(activeMasternodeInfo.blsPubKeyOperator == nullptr);
-    }
-    fMasternodeMode = false;
```

Comment:
It should be totally fine; as the call to LLMQContext::Start is also in AppInitMain; just a few hundred lines later: https://github.com/dashpay/dash/blob/d58e41514d7d93c61cbcd237c855674a87798f38/src/init.cpp#L1959;|

also my guess is that @kwvg was forced to due this by compiler / linter

Reply:
The code had to be moved somewhere.

Initialization of `activeMasternodeInfo` and `activeMasternodeManager` were disjoint since they were globals in their own right but since `activeMasternodeInfo`'s contents were being subsumed into `activeMasternodeManager`, the state needed to be initialized after the manager (and later commits remove later initialization by adding it into the manager's constructor).

It was moved downwards instead of upwards because downwards was where the manager was being initialized and to avoid upsetting anything it was kept that way. As a bonus, keeping it downward meant I get to keep passing a dereferenced `node.connman`.

Is it possible that bugs may show up but the node doesn't start until much later and aside from `CDKGSessionManager`, I'm not aware of any constructors that check on "masternode mode" status.